trieste/models/gpflux/sampler.py

```diff
@@ -54,4 +74,122 @@ def __call__(self, X: TensorType) -> tf.Tensor:
                 X = f(X)
             return X
 
-    return ChainedSample()
+    return ChainedSample().__call__
+
+
+class DeepGaussianProcessTrajectorySampler(TrajectorySampler[GPfluxPredictor]):
+    r"""
+    This sampler provides trajectory samples from a :class:`GPfluxPredictor`\ 's predictive
+    distribution, for :class:`GPfluxPredictor`\s with an underlying gpflux.models.DeepGP model.
+    """
+
+    def __init__(self, model: GPfluxPredictor):
+        """
+        :param model: The model to sample from. Must be a :class:`GPfluxPredictor` with an
+            an underlying gpflux.models.DeepGP `model_gpflux`.
+        :raise ValueError: If model is not a :class:`GPfluxPredictor`, or the underlying
+            `model_gpflux` is not a gpflux.models.DeepGP
+        """
+        if not isinstance(model, GPfluxPredictor):
+            raise ValueError(
+                f"Model must be a gpflux.interface.GPfluxPredictor, received {type(model)}"
+            )
+
+        super().__init__(model)
+
+        self.model = model.model_gpflux
```

Comment:
why not private?
```suggestion
        self._model = model.model_gpflux
```

Reply:
`TrajectorySampler` (the superclass here) already has a `self._model` which is required to be a `ProbabilisticModelType`, but yes I agree it should be renamed and private.